### PR TITLE
build system overhaul party PR

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
This is never going to land.  I'll start breaking it out into smaller bits tomorrow.

### CODE REVIEW

File: pkg/lib/stampfile-plugin.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                 *   /build/dir/dist/base1
+                 *
+                 * Let's put a stamp there.
+                 */
+                stamps[dir + '/stamp'] = true;
+            }
+
+            for (const stampfile in stamps) {
+                fs.writeFileSync(stampfile, '');
+            }
```
Comment (martinpitt) [inline pkg/lib/stampfile-plugin.js:41]: Wouldn't it be simpler to directly call writeFileSync() in the first loop, and drop the `stamps` object?

**Review by martinpitt — COMMENTED** [review]
I have some trouble to see where this is going. It feels like having a new plugin that just writes stamp files is a bit overkill? E.g. eventually we want to move building of manifest.json to webpack, and then the manifests would IMHO be a more appropriate "stamp", and we could just ditch the stamp files completely. Building manifests is probably also going to be a plugin, but I don't think we want two?

File: pkg/lib/stampfile-plugin.js (hunk 1 above)
Comment (allisonkarlitskaya) [reply pkg/lib/stampfile-plugin.js:41]: The first loop is going to encounter very many files within this subdirectory, so we'd end up writing the stamp file over and over again...

Comment (allisonkarlitskaya) [reply pkg/lib/stampfile-plugin.js:41]: The comment about the manifest.json acting as a stamp is reasonable...  we used to use the Makefile.deps for this purpose, in fact...

Comment (allisonkarlitskaya) [reply pkg/lib/stampfile-plugin.js:41]: For the record about "where this is going": the idea is eventually that typing `webpack` will produce exactly the same output in `dist/` as if we typed `make`.

Comment (martinpitt) [reply pkg/lib/stampfile-plugin.js:41]: Loop> make sense, sorry for misreading that.

That's a good goal -- I rather meant whether the stamp file plugin is meant to be a temporary thing or a lasting (part of) the final solution.

**Review by martinpitt — APPROVED** [review]
OK, let's go with that for now, if it helps you to break the webpack cleanups into smaller steps.  I can still bicker about "stamp and manifest plugin are 90% identical" later on, and we can simplify things once everything is working. Thanks!

**Comment by allisonkarlitskaya** [comment]
Most of the important bits of this landed elsewhere.  All that's really left are the trivial changes to `WEBPACK_INPUTS` and `WEBPACK_OUTPUTS`.  Those can be handled elsewhere, since this PR is a big and rather outdated mess.

**Pull request closed without merging**